grf 2.0 Cython compatibility

Hey @erikcs . Thanks for pointing out the 2.0 release with changes. I've begun updating skgrf and for the most part it has been pretty straightforward. The updates to the Data class have been beneficial (I'm now able to run without using a special C++ wrapper for numpy), and the changes have been minimal with respect to the bindings interface. There is one small issue that is preventing me from upgrading:

You can see from this [CI run](https://github.com/crflynn/skgrf/pull/61/checks?check_run_id=2948984358) that Cython fails to compile the python extension because of a variable name collision in the grf namespace in two of the strategy files. I believe this is because Cython requires includes of cpp files in addition to headers for C++ extensions, whereas internally here I think you would only include header files. When I change the name of one of these variables locally, it compiles fine, and all of the tests pass.

Here is the Cython compilation output indicating the error.
```
      In file included from skgrf/grf.cpp:949:
      skgrf/./grf/src/prediction/InstrumentalPredictionStrategy.cpp:35:19: error: redefinition of ‘const size_t grf::NUM_TYPES’
         35 | const std::size_t NUM_TYPES = 7;
            |                   ^~~~~~~~~
      In file included from skgrf/grf.cpp:943:
      skgrf/./grf/src/prediction/CausalSurvivalPredictionStrategy.cpp:30:19: note: ‘const size_t grf::NUM_TYPES’ previously defined here
         30 | const std::size_t NUM_TYPES = 2;
            |                   ^~~~~~~~~
      error: command 'gcc' failed with exit status 1
```

Now, I'm not a Cython or C++ expert, but I've tried various things like excluding cpp files from cython extension includes, but I don't think it's possible to build an extension without them. If it's not too much trouble I'd like to request modifying these variable names, and making a patch release. This should enable compilation using Cython.

--- a/core/src/prediction/CausalSurvivalPredictionStrategy.cpp
+++ b/core/src/prediction/CausalSurvivalPredictionStrategy.cpp
@@ -26,8 +26,7 @@ namespace grf {
 
 const std::size_t CausalSurvivalPredictionStrategy::NUMERATOR = 0;
 const std::size_t CausalSurvivalPredictionStrategy::DENOMINATOR = 1;
-
-const std::size_t NUM_TYPES = 2;
+const std::size_t CausalSurvivalPredictionStrategy::NUM_TYPES = 2;
 
 size_t CausalSurvivalPredictionStrategy::prediction_length() const {
   return 1;

--- a/core/src/prediction/CausalSurvivalPredictionStrategy.h
+++ b/core/src/prediction/CausalSurvivalPredictionStrategy.h
@@ -33,6 +33,7 @@ class CausalSurvivalPredictionStrategy final: public OptimizedPredictionStrategy
 public:
   static const std::size_t NUMERATOR;
   static const std::size_t DENOMINATOR;
+  static const std::size_t NUM_TYPES;
 
   size_t prediction_value_length() const;
   PredictionValues precompute_prediction_values(

--- a/core/src/prediction/InstrumentalPredictionStrategy.cpp
+++ b/core/src/prediction/InstrumentalPredictionStrategy.cpp
@@ -31,8 +31,7 @@ const std::size_t InstrumentalPredictionStrategy::OUTCOME_INSTRUMENT = 3;
 const std::size_t InstrumentalPredictionStrategy::TREATMENT_INSTRUMENT = 4;
 const std::size_t InstrumentalPredictionStrategy::INSTRUMENT_INSTRUMENT = 5;
 const std::size_t InstrumentalPredictionStrategy::WEIGHT = 6;
-
-const std::size_t NUM_TYPES = 7;
+const std::size_t InstrumentalPredictionStrategy::NUM_TYPES = 7;
 
 size_t InstrumentalPredictionStrategy::prediction_length() const {
     return 1;

--- a/core/src/prediction/InstrumentalPredictionStrategy.h
+++ b/core/src/prediction/InstrumentalPredictionStrategy.h
@@ -38,6 +38,7 @@ public:
   static const std::size_t TREATMENT_INSTRUMENT;
   static const std::size_t INSTRUMENT_INSTRUMENT;
   static const std::size_t WEIGHT;
+  static const std::size_t NUM_TYPES;
 
   size_t prediction_value_length() const;
   PredictionValues precompute_prediction_values(

